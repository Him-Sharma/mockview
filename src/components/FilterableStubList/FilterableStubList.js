import React, { useState } from "react";
import Aux from "../../hoc/Aux";
import StubList from "./StubList/StubList";
import Search from "./Search/Search";

const FilterableStubList = (props) => {
  const [filterText, setFilterText] = useState("");

  return (
    <Aux>
      <Search filterText={filterText} onFilterTextChange={setFilterText} />
      <StubList stubs={props.stubs} filterText={filterText} />
    </Aux>
  );
};
export default FilterableStubList;
