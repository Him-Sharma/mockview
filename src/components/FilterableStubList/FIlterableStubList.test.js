import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import FilterableStubList from "./FilterableStubList";

afterEach(cleanup);
const stubs = [
  {
    predicates: [
      {
        equals: {
          method: "GET",
          path: "/posts",
        },
      },
    ],
    _links: {
      self: {
        href: "http://test",
      },
    },
  },
];
it("renders correctly", () => {
  const { baseElement } = render(<FilterableStubList stubs={stubs} />);
  expect(baseElement).toMatchSnapshot();
});
