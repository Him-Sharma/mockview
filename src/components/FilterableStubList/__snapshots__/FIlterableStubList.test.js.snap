// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <section
      class="search"
    >
      <div
        class="card"
      >
        <div
          class="search-input"
        >
          <label>
            Filter by Title
          </label>
          <input
            data-testid="filterTextInput"
            id="filterTextInput"
            type="text"
            value=""
          />
        </div>
      </div>
    </section>
    <section
      class="stub-list"
    >
      <ul>
        <li>
          <span>
            GET
          </span>
          <span>
            /posts
          </span>
        </li>
      </ul>
    </section>
  </div>
</body>
`;
